Dockerfile

```diff
@@ -27,70 +27,55 @@ ENV NUXT_TELEMETRY_DISABLED=1
 # build the application and generate a distribution package
 RUN pnpm run build
 
+
 # ==
-# development
+# pre-production
 # ==
-# application for development purposes
-FROM node:16 AS dev
-
-WORKDIR /usr/app
+# trash existing deps and install only necessary deps to reduce final image size
+FROM builder AS postbuild
 
-# Install pnpm
-RUN npm install -g pnpm
+RUN rm -rf node_modules
 
-ENV NODE_ENV=development
-ENV CYPRESS_INSTALL_BINARY=0
-
-# copy files from local machine
-COPY . /usr/app
-
-# disable telemetry when building the app
-ENV NUXT_TELEMETRY_DISABLED=1
+ENV NODE_ENV=production
 
-# install dependencies (development dependencies included)
-RUN pnpm install
+RUN pnpm install --frozen-lockfile --store-dir=./pnpm-store
 
-# expose port 8443
-EXPOSE 8443
+# Delete everything that isn't nuxt related
+RUN ls node_modules | grep -v nuxt | xargs rm -rf
```

Comment:
If you want to avoid any possible error if the `node_modules` doesn't exists or there are no `nuxt` files within, you can use `find` command which doesn't throw any error code if there are no files during the lookup.

```bash
find node_modules -type f -not -name '*nuxt*' -delete
```